src/ShearStressTransportEquationSystem.C

```diff
@@ -607,6 +622,119 @@ ShearStressTransportEquationSystem::compute_f_one_blending()
+  const auto& meta = meshInfo.meta();
+  const auto& ngpMesh = meshInfo.ngp_mesh();
+  const auto& fieldMgr = meshInfo.ngp_field_manager();
+
+  const double betaStar = realm_.get_turb_model_constant(TM_betaStar);
+
+  const auto& tkeNp1 =
+    fieldMgr.get_field<double>(tke_->mesh_meta_data_ordinal());
+  const auto& sdrNp1 =
+    fieldMgr.get_field<double>(sdr_->mesh_meta_data_ordinal());
+  auto& lt =
+    fieldMgr.get_field<double>(lt_->mesh_meta_data_ordinal());
+
+  const stk::mesh::Selector sel =
+    (meta.locally_owned_part() | meta.globally_shared_part()) &
+    (stk::mesh::selectField(*lt_));
+
+  nalu_ngp::run_entity_algorithm(
+    "SST::compute_lt", ngpMesh, stk::topology::NODE_RANK, sel,
+    KOKKOS_LAMBDA(const MeshIndex& mi) {
+      const double tke = tkeNp1.get(mi, 0);
+      const double sdr = sdrNp1.get(mi, 0);
+
+      const double l_t = stk::math::sqrt(tke)/(stk::math::pow(betaStar, .25)*sdr);
+      lt.get(mi, 0) = l_t;
+    });
+
+  lt.modify_on_device();
+}
+
+/** Compute mixing length and gamma fields for 2003 SST implementation
+ */
+void
+ShearStressTransportEquationSystem::compute_gamma()
+{
+  using MeshIndex = nalu_ngp::NGPMeshTraits<>::MeshIndex;
+
+  const auto& meshInfo = realm_.mesh_info();
+  const auto& meta = meshInfo.meta();
+  const auto& ngpMesh = meshInfo.ngp_mesh();
+  const auto& fieldMgr = meshInfo.ngp_field_manager();
+
+  const double betaStar = realm_.get_turb_model_constant(TM_betaStar);
+  const double betaOne = realm_.get_turb_model_constant(TM_betaOne);
+  const double betaTwo = realm_.get_turb_model_constant(TM_betaTwo);
+  const double gammaOne = realm_.get_turb_model_constant(TM_gammaOne);
+  const double gammaTwo = realm_.get_turb_model_constant(TM_gammaTwo);
+
+  const bool lengthScaleLimiter = realm_.solutionOptions_->lengthScaleLimiter_;
+  double l_e;
+  if (lengthScaleLimiter) { 
+    // calculate maximum length scale
+    const double earthAngularVelocity = realm_.solutionOptions_->earthAngularVelocity_;
+    const double pi = std::acos(-1.0);
+    const double latitude = realm_.solutionOptions_->latitude_*pi/180.0;
+    const double referenceVelocity = realm_.solutionOptions_->referenceVelocity_;
+    const double corfac = 2.0*earthAngularVelocity*std::sin(latitude);
+    l_e = .00027*referenceVelocity/corfac;
+  }
+
+  const auto& fOneBlend =
+    fieldMgr.get_field<double>(fOneBlending_->mesh_meta_data_ordinal());
+  const auto& lt =
+    fieldMgr.get_field<double>(lt_->mesh_meta_data_ordinal());
+  auto& gamma =
+    fieldMgr.get_field<double>(gamma_->mesh_meta_data_ordinal());
+
+  const stk::mesh::Selector sel =
+    (meta.locally_owned_part() | meta.globally_shared_part()) &
+    (stk::mesh::selectField(*gamma_)); 
+
+  nalu_ngp::run_entity_algorithm(
+    "SST::compute_gamma", ngpMesh, stk::topology::NODE_RANK, sel,
+    KOKKOS_LAMBDA(const MeshIndex& mi) {
+      const double f1 = fOneBlend.get(mi, 0); 
+      const double l_t = lt.get(mi, 0);
```

Comment:
I think you can move `l_t` into the `if (lengthScaleLimiter)` block

Reply:
Good catch; changed this.